src/kibana/components/config/defaults.js

```diff
@@ -43,6 +43,21 @@ define(function (require) {
       value: 6,
       description: 'The maximum geoHash size allowed in a tile map',
     },
+    'visualization:tileMap:wms': {
+      value: false,
+      description: 'Set for WMS mapping provider'
+    },
+    'visualization:tileMap:url': {
+      value: 'http://otile{s}.mqcdn.com/tiles/1.0.0/map/{z}/{x}/{y}.jpeg',
+      description: 'URL of mapping provider'
+    },
+    'visualization:tileMap:options': {
+      value: '{ "attribution": "Tiles by <a href=\\"http://www.mapquest.com/\\">MapQuest</a> &mdash; ' +
+        'Map data &copy; <a href=\\"http://openstreetmap.org\\">OpenStreetMap</a> contributors, ' +
+        '<a href=\\"http://creativecommons.org/licenses/by-sa/2.0/\\">CC-BY-SA</a>", "subdomains": "1234" }',
+      description: 'TileLayer options, see leafletjs documentation',
```

Comment:
Given the size and complexity of this I'd prefer to leave value undefined here and add defaults to the call to get.config. 

I'd also recommend updating config.get to check if the value is already a plain object before attempting to JSON parse it. That way you can pass an object to config.get as the default value